File: app/src/main/java/com/jiujiu/mvvmTemplate/ui/main/MainViewPagerAdapter.java
package com.jiujiu.mvvmTemplate.ui.main;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.jiujiu.mvvmTemplate.ui.customers.CustomersFragment;
import com.jiujiu.mvvmTemplate.ui.orders.OrdersFragment;
import com.jiujiu.mvvmTemplate.ui.products.ProductsFragment;

public class MainViewPagerAdapter extends FragmentPagerAdapter {

    public MainViewPagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return ProductsFragment.getInstance();
            case 1:
                return OrdersFragment.getInstance();
            case 2:
                return CustomersFragment.getInstance();
            default:
                throw new IllegalArgumentException("Error fragment in viewpager");
        }
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: app/src/main/java/com/jiujiu/mvvmTemplate/ui/main/MainViewPagerAdapter.kt
package com.jiujiu.mvvmTemplate.ui.main

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.jiujiu.mvvmTemplate.ui.customers.CustomersFragment
import com.jiujiu.mvvmTemplate.ui.orders.OrdersFragment
import com.jiujiu.mvvmTemplate.ui.products.ProductsFragment

class MainViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> ProductsFragment.getInstance()
            1 -> OrdersFragment.getInstance()
            2 -> CustomersFragment.getInstance()
            else -> throw IllegalArgumentException("Error fragment in viewpager")
        }
    }

    override fun getCount(): Int {
        return 3
    }
}
